src/main/java/org/terasology/launcher/util/DownloadUtils.java

```diff
@@ -10,53 +10,56 @@
 import java.io.BufferedInputStream;
 import java.io.BufferedOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.HttpURLConnection;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
 
 public final class DownloadUtils {
 
     private static final Logger logger = LoggerFactory.getLogger(DownloadUtils.class);
 
-    private static final int CONNECT_TIMEOUT = 1000 * 30;
-    private static final int READ_TIMEOUT = 1000 * 60 * 5;
+    private static final Duration CONNECT_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration READ_TIMEOUT = Duration.ofMinutes(5);
 
     private DownloadUtils() {
     }
 
-    public static void downloadToFile(URL downloadURL, Path file, ProgressListener listener) throws DownloadException {
+    public static CompletableFuture<Void> downloadToFile(URL downloadURL, Path file, ProgressListener listener) throws DownloadException {
         listener.update(0);
 
-        final HttpURLConnection connection = getConnectedDownloadConnection(downloadURL);
+        var result = getConnectedDownloadConnection(downloadURL);
```

Comment:
Hm, I either want an explicit type here, or a better name for `getConnectedDownloadConnection` :see_no_evil: